framework/src/predictors/AdamsPredictor.C

```diff
@@ -73,6 +73,10 @@ AdamsPredictor::apply(NumericVector<Number> & sln)
   // that gets called on time step begin.
   // That means that history control must go here.
   historyControl();
+
+  if (!shouldApply())
```

Comment:
I like `timestepSetup()` better for now... let's go that direction.

We can add an `initialize()` later if its needed.  `timestepSetup()` is even more standard across MOOSE.